command/tls_cert_create_test.go

```diff
@@ -149,14 +137,134 @@ func TestTlsCertCreateCommand_fileCreate(t *testing.T) {
 			nil,
 			"",
 		},
-		{"cli0-region2-altdomain",
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		require.True(t, t.Run(tc.name, func(t *testing.T) {
+			ui := cli.NewMockUi()
+			cmd := &TLSCertCreateCommand{Meta: Meta{Ui: ui}}
+			require.Equal(t, 0, cmd.Run(tc.args))
+			require.Equal(t, tc.errOut, ui.ErrorWriter.String())
+
+			// is a valid cert expects the cert
+			cert := testutil.IsValidCertificate(t, tc.certPath)
+			require.Equal(t, tc.expectCN, cert.Subject.CommonName)
+			require.True(t, cert.BasicConstraintsValid)
+			require.Equal(t, x509.KeyUsageDigitalSignature|x509.KeyUsageKeyEncipherment, cert.KeyUsage)
+			switch tc.typ {
+			case "server":
+				require.Equal(t,
+					[]x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+					cert.ExtKeyUsage)
+			case "client":
+				require.Equal(t,
+					[]x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+					cert.ExtKeyUsage)
+			case "cli":
+				require.Len(t, cert.ExtKeyUsage, 0)
+			}
+			require.False(t, cert.IsCA)
+			require.Equal(t, tc.expectDNS, cert.DNSNames)
+			require.Equal(t, tc.expectIP, cert.IPAddresses)
+		}))
+	}
+}
+
+func TestTlsCertCreateAlternateDomainCommand_fileCreate(t *testing.T) {
+	testDirAlt := t.TempDir()
+	previousDirectory, err := os.Getwd()
+	require.NoError(t, err)
+	require.NoError(t, os.Chdir(testDirAlt))
+	defer os.Chdir(previousDirectory)
+
+	ui := cli.NewMockUi()
+	caCmd := &TLSCACreateCommand{Meta: Meta{Ui: ui}}
+
+	// Setup CA keys
+	caCmd.Run([]string{"-name-constraint=true", "-domain=foo"})
+
+	type testcase struct {
+		name      string
+		typ       string
+		args      []string
+		certPath  string
+		keyPath   string
+		expectCN  string
+		expectDNS []string
+		expectIP  []net.IP
+		errOut    string
+	}
+
+	// The following subtests must run serially.
+	cases := []testcase{
+		{"server0-altdomain",
+			"server",
+			[]string{"-server", "-domain", "foo"},
+			"global-server-foo.pem",
+			"global-server-foo-key.pem",
+			"server.global.foo",
+			[]string{
+				"server.global.foo",
+				"server.global.nomad",
+				"localhost",
+			},
+			[]net.IP{{127, 0, 0, 1}},
+			"==> WARNING: Server Certificates grants authority to become a\n    server and access all state in the cluster including root keys\n    and all ACL tokens. Do not distribute them to production hosts\n    that are not server nodes. Store them as securely as CA keys.\n",
+		},
```

Comment:
It might be a good idea just to split these cases out into unit tests of `recordPreparation` rather than having to exercise the whole thing.

Reply:
Done